src/plugins/kibana_react/public/app_links/click_handler.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { ApplicationStart } from 'src/core/public';
+import { getClosestLink, hasActiveModifierKey } from './utils';
+
+interface CreateCrossAppClickHandlerOptions {
+  container: HTMLElement;
+  navigateToUrl: ApplicationStart['navigateToUrl'];
+}
+
+export const createCrossAppClickHandler = ({
+  container,
+  navigateToUrl,
+}: CreateCrossAppClickHandlerOptions): React.MouseEventHandler<HTMLElement> => {
+  return (e) => {
+    if (container == null) {
+      return;
+    }
+    // see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12239
+    const target = e.target as HTMLElement;
+
+    const link = getClosestLink(target, container);
```

Comment:
Not sure, due to browser support: https://caniuse.com/#search=closest?
Unfortunately I don't have a browserstack account to ensure this API is properly shimmed by `corejs`

Do we have a usable account somewhere?

Reply:
We are discounting IE11 support in v7.9, so it's not a problem.

Reply:
That requires quite a bit of changes in the handler unit tests as it can no longer pass down the container for testing, but I guess using native APIs should be encouraged.

Reply:
Hum. it seems that our version of `jsdom` doesnt supports `closest`, which is strange as it's supposed to be available for quite a long time now.

```ts
    TypeError: target.closest is not a function

      41 |     const target = e.target as HTMLElement;
      42 |
    > 43 |     const link = target.closest(`.${containerClass} a`) as HTMLAnchorElement;
```

```ts
console.log(target, target.parentElement, target.closest, target.parentElement.closest)

// HTMLAnchorElement {} HTMLDivElement {...} undefined undefined
```

Reply:
Confirmed: the `jsdom` version jest is using (`node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/nodes/Element-impl.js`) is a `11.5.1`, which does not implements `closest`...

Reply:
That's a shame. Let's keep the current implementation then. We can add a comment to switch to native API later.

Reply:
Will add a comment for when https://github.com/elastic/kibana/pull/58095 lands.

Reply:
Unfortunately, I wasn't able to bring the most recent version of `jsdom` in that PR, because seems a combination of `jest@25` with `jsdom>=13` got serious performance degradation and we cannot bump to `jest@26`, because it requires `typescript>=3.8`, so I hope @restrry PR gets merged soon and I will give it a try to `jest@26` and `jsdom@16`